helix-core/src/main/java/org/apache/helix/common/caches/BasicClusterDataCache.java

```diff
@@ -35,11 +34,17 @@
 /**
  * Cache the basic cluster data, including LiveInstances, InstanceConfigs and ExternalViews.
  */
-public class BasicClusterDataCache {
+public class BasicClusterDataCache implements ControlContextProvider {
   private static Logger LOG = LoggerFactory.getLogger(BasicClusterDataCache.class.getName());
 
-  protected Map<String, LiveInstance> _liveInstanceMap;
-  protected Map<String, InstanceConfig> _instanceConfigMap;
+  private static final String INSTANCE_CONFIG = "InstanceConfig";
+  private static final String LIVE_INSTANCE = "LiveInstance";
+
+  private String _clusterEventId;
+  private String _pipelineName;
```

Comment:
This can be static final, I think.

Reply:
How about we just return `AbstractDataCache.UNKNOWN_PIPELINE` in `getPipelineName()` because `AbstractDataCache.UNKNOWN_PIPELINE` is already static final? We don't have create another duplicate static final.